xhttp/client.go

```diff
@@ -158,7 +159,22 @@ func (r *retrierClient) do(ctx context.Context, req *http.Request, requestBody [
 		}
 		log.Debug("xhttp.Client: retrying request with error status code")
 		r.onRetry(req, res, err)
-		// Maybe add handling for Retry-After header, so far this seems to be enough
+
+		// handle Retry-After header
+		retryAfter := res.Header.Get("Retry-After")
+		requestedDuration, requestedTime, err := ParseRetryAfter(retryAfter)
+		switch {
+		case err != nil:
+			log.Warn(fmt.Sprintf("xhttp.Client: %v", err))
+		case requestedDuration != 0:
+			log.Debug("xhttp.Client: following Retry-After header", "duration", requestedDuration)
+			sleepPeriod = requestedDuration
+		case !requestedTime.IsZero():
+			sleepPeriod = time.Until(requestedTime)
```

Comment:
What if the sleep period is negative ? (the requested time is on the past). I know it would be some problem/bug on the server side but that would cause us to start sending requests as fast as possible on each try. Actually it could also be a bug/issue with clock synchronization on our side, making our time.Now to be way in the future, etc. IMHO when for any reason this is smaller than 1 second (it is the smallest retry period on the header when you use the duration on the header) we should use the configured sleep period and assume there is something wrong with the headers/clocks (wall clocks are nasty... that is why usually the duration is a better idea..but the header support both).

Reply:
Having said that...the same should be done for negative durations...it doesn't make sense but on the web you just expect the worse =P..in this case the error is solely on the server side...but would be better to handle it gracefully.

Reply:
I added a `minSleepPeriod` to avoid those cases